Sources/OTel/Metrics/OTelMetricProducer/OTLPMetricsDataModel.swift

```diff
@@ -0,0 +1,150 @@
+///
+/// The types in this file represent the subset of the OTLP datamodel that we use, which map over the protobuf types.
+
+@_spi(Metrics)
+public struct OTelResourceMetrics: Equatable, Sendable {
+    public var resource: OTelResource?
+    public var scopeMetrics: [OTelScopeMetrics]
+}
+
+@_spi(Metrics)
+public struct OTelScopeMetrics: Equatable, Sendable {
+    public var scope: OTelInstrumentationScope?
+    public var metrics: [OTelMetricPoint]
+}
+
+@_spi(Metrics)
+public struct OTelInstrumentationScope: Equatable, Sendable {
+    public var name: String?
+    public var version: String?
+    public var attributes: [OTelAttribute]
+    public var droppedAttributeCount: Int32
+}
+
+/// - Seealso: [](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.29.0/specification/metrics/data-model.md#metric-points)
+@_spi(Metrics)
+public struct OTelMetricPoint: Equatable, Sendable {
+    public var name: String
+    public var description: String
+    public var unit: String
+    public enum OTelMetricData: Equatable, Sendable {
+        case gauge(OTelGauge)
+        case sum(OTelSum)
+        case histogram(OTelHistogram)
+    }
+
+    public var data: OTelMetricData
+}
+
+/// - Seealso: [](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.29.0/specification/metrics/data-model.md#sums)
+@_spi(Metrics)
+public struct OTelSum: Equatable, Sendable {
+    public var points: [OTelNumberDataPoint]
+    public var aggregationTemporality: OTelAggregationTemporailty
+    public var monotonic: Bool
+}
+
+/// - Seealso: [](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.29.0/specification/metrics/data-model.md#gauge)
+@_spi(Metrics)
+public struct OTelGauge: Equatable, Sendable {
+    public var points: [OTelNumberDataPoint]
+}
+
+/// - Seealso: [](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.29.0/specification/metrics/data-model.md#histogram)
+@_spi(Metrics)
+public struct OTelHistogram: Equatable, Sendable {
+    public var aggregationTemporality: OTelAggregationTemporailty
+    public var points: [OTelHistogramDataPoint]
+}
+
+/// - Seealso: [](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.29.0/specification/metrics/data-model.md)
+@_spi(Metrics)
+public struct OTelAttribute: Hashable, Equatable, Sendable {
+    public var key: String
+    public var value: String
+}
+
+/// - Seealso: [](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.29.0/specification/metrics/data-model.md#temporality)
+@_spi(Metrics)
+public enum OTelAggregationTemporailty: Equatable, Sendable {
+    case delta
+    case cumulative
+}
+
+@_spi(Metrics)
+public struct OTelNumberDataPoint: Equatable, Sendable {
+    public var attributes: [OTelAttribute]
+    public var startTimeNanosecondsSinceEpoch: UInt64?
+    public var timeNanosecondsSinceEpoch: UInt64
+    public enum OTelPointValue: Equatable, Sendable {
```

Comment:
I think we can drop the `OTel` prefix here since it's nested in a type that already has the prefix.